test/parallel/test-buffer-alloc.js

```diff
@@ -0,0 +1,1466 @@
+  const size = buffer.write(testValue, 0, 'utf8');
+  console.log('bytes written to buffer: ' + size);
+  const slice = buffer.toString('utf8', 0, size);
+  assert.equal(slice, testValue);
+}
+
+{
+  // Test triple  slice
+  const a = Buffer.allocUnsafe(8);
+  for (let i = 0; i < 8; i++) a[i] = i;
+  const b = a.slice(4, 8);
+  assert.equal(4, b[0]);
+  assert.equal(5, b[1]);
+  assert.equal(6, b[2]);
+  assert.equal(7, b[3]);
+  const c = b.slice(2, 4);
+  assert.equal(6, c[0]);
+  assert.equal(7, c[1]);
+}
+
+{
+  const d = Buffer.from([23, 42, 255]);
+  assert.equal(d.length, 3);
+  assert.equal(d[0], 23);
+  assert.equal(d[1], 42);
+  assert.equal(d[2], 255);
+  assert.deepEqual(d, Buffer.from(d));
+}
+
+{
+  const e = Buffer.from('über');
+  console.error('uber: \'%s\'', e.toString());
+  assert.deepEqual(e, Buffer.from([195, 188, 98, 101, 114]));
+}
+
+{
+  const f = Buffer.from('über', 'ascii');
+  console.error('f.length: %d     (should be 4)', f.length);
+  assert.deepEqual(f, Buffer.from([252, 98, 101, 114]));
+}
+
+['ucs2', 'ucs-2', 'utf16le', 'utf-16le'].forEach(function(encoding) {
+  {
+    const f = Buffer.from('über', encoding);
+    console.error('f.length: %d     (should be 8)', f.length);
+    assert.deepEqual(f, Buffer.from([252, 0, 98, 0, 101, 0, 114, 0]));
+  }
+
+  {
+    const f = Buffer.from('привет', encoding);
+    console.error('f.length: %d     (should be 12)', f.length);
+    assert.deepEqual(f,
+      Buffer.from([63, 4, 64, 4, 56, 4, 50, 4, 53, 4, 66, 4]));
+    assert.equal(f.toString(encoding), 'привет');
+  }
+
+  {
+    const f = Buffer.from([0, 0, 0, 0, 0]);
+    assert.equal(f.length, 5);
+    const size = f.write('あいうえお', encoding);
+    console.error('bytes written to buffer: %d     (should be 4)', size);
+    assert.equal(size, 4);
+    assert.deepEqual(f, Buffer.from([0x42, 0x30, 0x44, 0x30, 0x00]));
+  }
+});
+
+{
+  const f = Buffer.from('\uD83D\uDC4D', 'utf-16le'); // THUMBS UP SIGN (U+1F44D)
+  assert.equal(f.length, 4);
+  assert.deepEqual(f, Buffer.from('3DD84DDC', 'hex'));
+}
+
+
+var arrayIsh = {0: 0, 1: 1, 2: 2, 3: 3, length: 4};
+var g = Buffer.from(arrayIsh);
+assert.deepEqual(g, Buffer.from([0, 1, 2, 3]));
+var strArrayIsh = {0: '0', 1: '1', 2: '2', 3: '3', length: 4};
+g = Buffer.from(strArrayIsh);
+assert.deepEqual(g, Buffer.from([0, 1, 2, 3]));
+
+
+//
+// Test toString('base64')
+//
+assert.equal('TWFu', (Buffer.from('Man')).toString('base64'));
+
+{
+  // test that regular and URL-safe base64 both work
+  const expected = [0xff, 0xff, 0xbe, 0xff, 0xef, 0xbf, 0xfb, 0xef, 0xff];
+  assert.deepEqual(Buffer.from('//++/++/++//', 'base64'),
+                   Buffer.from(expected));
+  assert.deepEqual(Buffer.from('__--_--_--__', 'base64'),
+                   Buffer.from(expected));
+}
+
+{
+  // big example
+  const quote = 'Man is distinguished, not only by his reason, but by this ' +
+              'singular passion from other animals, which is a lust ' +
```

Comment:
This was copied directly from test-buffer.js which has the same issue. Let's fix this one in a separate PR